### PR TITLE
refactor(tooling): Switch to Next.js app router

### LINKED ISSUE #112: 🛠 Tooling: Switch to Next.js app router
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/OpenContributionsProject/opencontributionssite/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This repo is set up with [Next.js `pages/`](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts). Which is the classic/traditional way of doing app pages in Next.js.

But Next.js has a shiny new way of doing things now! [Next.js `app/` routing](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts) is awesome and a better way to do things.

Let's migrate this project do `app/`!

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
## PR Checklist

- [x] Addresses an existing open issue: fixes #112
- [x] That issue was marked as [accepting prs](https://github.com/OpenContributionsProject/opencontributionssite/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/OpenContributionsProject/opencontributionssite/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Switch from pages to app router

### CODE REVIEW

**Comment by blackgirlbytes** [comment]
@kevinzunigacuellar not you being amazing! Thank you for this PR!!! I'll try to review it today!

**Comment by kevinzunigacuellar** [comment]
I am not quite sure how to fix the issues with `ts-prune` since most of these are nextjs convention related.

File: next.config.mjs [hunk 1]
```diff
@@ -11,7 +11,11 @@ const withMDX = nextMdx({
...
-	pageExtensions: ["js", "jsx", "md", "mdx", "ts", "tsx"],
+const nextConfig = {
 	reactStrictMode: true,
-});
+	experimental: {
+		mdxRs: true,
+	},
+};
+
+export default withMDX(nextConfig);
```
Comment (JoshuaKGoldberg) [inline next.config.mjs:21]: [Nit] Is there a reason to move from an inline object inside the `export default withMdx(...)` to a separate variable? Not a blocker in my opinion, just curious 😄 _(with a mild preference for the older way)_

File: src/app/about/page.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import About from "./about.mdx";
+
+export default function AboutPage() {
+	return <About />;
+}
```
Comment (JoshuaKGoldberg) [inline src/app/about/page.tsx:5]: [Question] Huh, does the new App router not support `.mdx` directly? 

I suppose this isn't a blocker (I can't figure out a direct `page.mdx` or whatnot locally), but a bit of an "ah well".

Docs link I could find: https://nextjs.org/docs/app/building-your-application/configuring/mdx

File: src/app/get-involved/get-involved.mdx [hunk 3]
```diff
@@ -16,7 +16,7 @@ See our [_area: data gathering_ issues](https://github.com/OpenContributionsProj
 
 The site is a little visually bland right now.
 We're going to need to make it presentable, with a full site theme and backing design system.
-See our [_area: design_ issues](https://github.com/OpenContributionsProject/opencontributionssite/labels/area%3A%20data%20gathering) for ones you can get started on.
+See our [_area: design_ issues](https://github.com/OpenContributionsProject/opencontributionssite/labels/area%3A%20design) for ones you can get started on.
```
Comment (JoshuaKGoldberg) [inline src/app/get-involved/get-involved.mdx:19]: Oh ha yes...

File: tsconfig.json [hunk 4]
```diff
@@ -13,8 +13,13 @@
 		"resolveJsonModule": true,
 		"skipLibCheck": true,
 		"strict": true,
-		"target": "ESNext"
+		"target": "ESNext",
+		"plugins": [
+			{
+				"name": "next"
+			}
```
Comment (JoshuaKGoldberg) [inline tsconfig.json:20]: [Question] Ooh I haven't seen this before and can't find docs on it. Is this described anywhere?

I see that `next build` and `next dev` add it automatically, so I guess we need it 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
No blockers from me, this is great! Just a few questions because I don't know all these things 😄 thanks! 🔥

**Comment by JoshuaKGoldberg** [comment]
> I am not quite sure how to fix the issues with ts-prune

You'll want to update `.ts-prunerc` to include `src/app` instead of `src/pages`. `lint:prune` calls to [`ts-prune`](https://github.com/nadeesha/ts-prune), which is configured by that file.

File: src/app/about/page.tsx (hunk 2 above)
Comment (kevinzunigacuellar) [reply src/app/about/page.tsx:5]: It seems that it doesn't according to the docs. We can revisit this later I guess 😇

File: tsconfig.json (hunk 4 above)
Comment (kevinzunigacuellar) [reply tsconfig.json:20]: yes, this was added automagically. I have no idea 😆 tbh

File: next.config.mjs (hunk 1 above)
Comment (kevinzunigacuellar) [reply next.config.mjs]: the older way it is 👴 [cb34e5e](https://github.com/OpenContributionsProject/opencontributionssite/pull/114/commits/cb34e5ea040d4f9390dd6f3ea6067881fe940fa5)

**Review by blackgirlbytes — APPROVED** [review]

**Comment by blackgirlbytes** [comment]
I'm gonna merge. Hope that's okay!

**Comment by JoshuaKGoldberg** [comment]
Rick Flag going WHOOOO

**Comment by blackgirlbytes** [comment]
@allcontributors add @kevinzunigacuellar for tooling

**Pull request merged**